internal/core/workers/operation_execution_worker/metrics.go

```diff
@@ -0,0 +1,105 @@
+package operation_execution_worker
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/topfreegames/maestro/internal/core/monitoring"
+)
+
+const (
+	LabelNoOperationExecutorFound = "no_operation_executor_found"
+	LabelShouldNotExecute         = "should_not_execute"
+	LabelNextOperationFailed      = "next_operation_failed"
+	LabelStartOperationFailed     = "start_operation_failed"
+)
+
+var (
+	operationExecutionLatencyMetric           *prometheus.HistogramVec
+	operationOnErrorLatencyMetric             *prometheus.HistogramVec
+	operationEvictedCountMetric               *prometheus.CounterVec
+	operationExecutionWorkerFailedCountMetric *prometheus.CounterVec
+)
+
+func init() {
```

Comment:
Why do we need to set the metrics on the package init?

Reply:
We could do it on the CMD as well, but once we don't have it, I believe this is the static right place to initialize them.

Also, in order to reset metrics (unit tests) we need to call the resetMetrics function.

Reply:
My question was more related to why not define them as package variables (since we don't need to check for their failure or perform any extra step). Is the only reason because we are asserting them on our unit tests? If yes, it might be related to [a point I raised on another PR](https://github.com/topfreegames/maestro/pull/155#discussion_r673934833), where we could have trouble relying on "global state", in this case, I think we need to come up with a solution for it instead of adding extra logic to our package, WDYT?

Reply:
Got it, that's the problem on setting metrics on global state and testing them on unit tests that does not clears the global state.

To me, we never came up before with this problem because no repo is testing metrics the way we're. Even prometheus testutils is not enough or does not give us a complete set of tools to test or even assert our metrics.

Totally agree we're relying on global state to fetch our metrics, still I cannot see an alternative given the premises:
* Keep metrics on "global state";
* Assert metrics on test;

The very use of promauto blocks the right assertion once it registers the metric directly to the default register/gather/collector. That lives on global scope.

What other alternatives would you suggest?